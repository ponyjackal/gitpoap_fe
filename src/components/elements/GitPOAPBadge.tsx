import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { TextLight, MidnightBlue } from '../../colors';

type Props = {
  imgUrl: string;
};

type HexProps = {
  size: number;
  borderSize: number;
};

const Hexagon = styled.div`
  clip-path: polygon(5% 25%, 50% 0, 95% 25%, 95% 75%, 50% 100%, 5% 75%);
`;

const HexOuterBorder = styled(Hexagon)<HexProps>`
  position: relative;
  --s: ${(props) => rem(props.size + 4 * props.borderSize)};
  width: var(--s);
  height: calc(var(--s) * 1);
  background-color: ${TextLight};
`;
const HexInnerBorder = styled(Hexagon)<HexProps>`
  --s: ${(props) => rem(props.size + 2 * props.borderSize)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  background: ${MidnightBlue};
`;

const HexBadge = styled(Hexagon)<{ url: string } & HexProps>`
  --s: ${(props) => rem(props.size)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  display: inline-block;
  font-size: initial; /* we reset the font-size if we want to add some content */
  background: url('${(props) => props.url}') no-repeat center center;
  background-size: cover;
`;

export const GitPOAPBadge = ({ imgUrl }: Props) => {
  const size = 200;
  const borderSize = 4;
  return (
    <HexOuterBorder size={size} borderSize={borderSize}>
      <HexInnerBorder size={size} borderSize={borderSize}>
        <HexBadge url={imgUrl} size={size} borderSize={borderSize} />
      </HexInnerBorder>
    </HexOuterBorder>
  );
};

/* Another option is to use SVG ~ https://stackoverflow.com/questions/67458234/mask-image-with-svg-shape-and-add-a-border*/
