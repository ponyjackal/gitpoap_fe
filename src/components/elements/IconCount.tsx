import styled from 'styled-components';
import { rem } from 'polished';
import { ExtraHover, ExtraPressed, TextGray, TextLight } from '../../colors';

type Props = {
  icon: React.ReactNode;
  href?: string;
  count: number;
};

const Count = styled.span`
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(12)};
  line-height: ${rem(17)};
  letter-spacing: ${rem(0.5)};
  text-transform: uppercase;
  color: ${TextLight};
  margin-left: ${rem(6)};
`;

const Container = styled.a`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  path {
    transition: 200ms fill ease;
    fill: ${TextGray};
  }

  &:hover {
    path {
      fill: ${ExtraHover};
    }
  }
  &:active {
    path {
      fill: ${ExtraPressed};
    }
  }
`;

export const IconCount = ({ icon, href, count }: Props) => {
  return (
    <Container href={href}>
      {icon}
      {count && <Count>{count}</Count>}
    </Container>
  );
};
