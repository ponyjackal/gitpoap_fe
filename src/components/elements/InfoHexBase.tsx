import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel } from '../../colors';

type Props = {
  children: React.ReactNode;
};

const StyledInfoHex = styled.div`
  display: inline-flex;
  flex-direction: column;
  min-width: ${rem(250)};
`;

const UpperTriangle = styled.div`
  clip-path: polygon(50% 0%, 0% 100%, 100% 100%);
  width: 100%;
  height: ${rem(70)};
  background-color: ${BackgroundPanel};
`;

const LowerTriangle = styled(UpperTriangle)`
  clip-path: polygon(50% 100%, 0 0, 100% 0);
`;

const Body = styled.div`
  background-color: ${BackgroundPanel};
  min-height: ${rem(100)};
`;

export const InfoHexBase = (props: Props) => {
  return (
    <StyledInfoHex>
      <UpperTriangle />
      <Body>{props.children}</Body>
      <LowerTriangle />
    </StyledInfoHex>
  );
};
