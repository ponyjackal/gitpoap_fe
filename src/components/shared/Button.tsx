import React from 'react';
import styled from 'styled-components';
import { Button as MButton } from '@mantine/core';
import { rem } from 'polished';
import { PrimaryBlue, TextAccent } from '../../colors';

type Props = {
  children: React.ReactNode;
  className?: string;
  onClick?: React.MouseEventHandler;
  disabled?: boolean;
  leftIcon?: React.ComponentProps<typeof MButton>['leftIcon'];
};

const StyledButton = styled(MButton)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${PrimaryBlue};
  font-family: 'PT Mono', monospace;
  font-weight: 700;
  color: white;
  text-transform: uppercase;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  transition: 200ms background ease;
  border-radius: ${rem(6)};
  padding: ${rem(8)} ${rem(14)};
  height: auto;

  &:hover {
    box-shadow: none;
    background-color: ${TextAccent};
  }

  &:active {
    background-color: #20457e;
  }
`;

export const Button = ({ children, className, onClick, disabled, leftIcon }: Props) => {
  return (
    <StyledButton
      compact
      className={className}
      size="md"
      disabled={disabled}
      onClick={onClick}
      leftIcon={leftIcon}
    >
      {children}
    </StyledButton>
  );
};
