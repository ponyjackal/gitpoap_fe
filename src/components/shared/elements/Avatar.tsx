import styled from 'styled-components';
import { rem } from 'polished';

export const Avatar = styled.img`
  height: ${rem(80)};
  width: ${rem(80)};
  border-radius: 50%;
`;
