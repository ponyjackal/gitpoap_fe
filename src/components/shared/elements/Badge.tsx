import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel3, TextLight } from '../../../colors';

export const Badge = styled.div`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: ${rem(0.5)} ${rem(4)};
  background: ${BackgroundPanel3};
  border-radius: ${rem(4)};

  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(11)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(1.2)};
  text-transform: uppercase;
  color: ${TextLight};
`;
