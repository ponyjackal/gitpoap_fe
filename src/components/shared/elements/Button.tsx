import React from 'react';
import styled from 'styled-components';
import { Button as ButtonUI, SharedButtonProps, ButtonProps } from '@mantine/core';
import { rem } from 'polished';
import { PrimaryBlue, TextGray, ExtraHover, ExtraPressed, DarkGray } from '../../../colors';

type Props = SharedButtonProps & {
  className?: string;
  onClick?: React.MouseEventHandler;
  onMouseEnter?: React.MouseEventHandler;
  onMouseLeave?: React.MouseEventHandler;
  disabled?: boolean;
  variant?: 'filled' | 'outline';
};

const StyledButton = styled(ButtonUI)<ButtonProps<'button'>>`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  font-family: 'PT Mono', monospace;
  font-weight: 700;
  text-transform: uppercase;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  transition: 150ms background ease, 150ms color ease, 150ms border ease;
  border-radius: ${rem(6)};
  padding: ${rem(8)} ${rem(14)};
  height: fit-content;

  &.mantine-Button-filled {
    background-color: ${PrimaryBlue};
    border: none;
    color: white;
    &:hover:not(:disabled) {
      background-color: ${ExtraHover};
    }
    &:active:not(:disabled) {
      background-color: ${ExtraPressed};
    }
    &:disabled {
      background-color: ${DarkGray};
      .mantine-Button-label {
        color: ${TextGray};
      }
    }
    &.mantine-Button-loading {
      background-color: ${PrimaryBlue};
      .mantine-Button-label {
        color: white;
      }
    }
  }

  &.mantine-Button-outline {
    border: ${rem(2)} solid ${TextGray};
    color: white;
    background-color: transparent;
    padding: ${rem(6)} ${rem(12)}; /* Adjusted for border */
    &:hover:not(:disabled) {
      border-color: ${ExtraHover};
      color: ${ExtraHover};
    }
    &:active:not(:disabled) {
      border-color: ${ExtraPressed};
      color: ${ExtraPressed};
    }
    &:disabled {
      background-color: transparent;
      border-color: ${TextGray};
      .mantine-Button-label {
        color: ${TextGray};
      }
    }
    &.mantine-Button-loading {
      &:before {
        top: ${rem(-2)};
        left: ${rem(-2)};
        right: ${rem(-2)};
        bottom: ${rem(-2)};
      }
      .mantine-Button-label {
        color: white;
      }
    }
  }
`;

export const Button = (props: Props) => {
  return <StyledButton {...props}>{props.children}</StyledButton>;
};
