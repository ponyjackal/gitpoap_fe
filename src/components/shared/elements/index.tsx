export * from './Avatar';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './ClaimCircle';
export * from './CollapsibleAddress';
export * from './CopyableText';
export * from './DateInput';
export * from './Divider';
export * from './GitPOAPBadge';
export * from './Header';
export * from './HexagonPath';
export * from './IconCount';
export * from './InfoHexBase';
export * from './Input';
export * from './InputWrapper';
export * from './Label';
export * from './Loader';
export * from './MultiSelect';
export * from './NavLink';
export * from './NumberInput';
export * from './OrgName';
export * from './Pagination';
export * from './POAPBadge';
export * from './Radio';
export * from './RelativeDate';
export * from './Select';
export * from './Share';
export * from './Skeletons';
export * from './Text';
export * from './TextArea';
export * from './Title';
export * from './TwitterShareButton';
