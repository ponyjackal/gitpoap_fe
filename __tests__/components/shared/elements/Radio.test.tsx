import { Radio } from '@mantine/core';
import 'jest-styled-components';
import { RadioGroup } from '../../../../src/components/shared/elements';
import { renderWithTheme } from '../../../__utils__/renderWithTheme';

describe('Radio', () => {
  it('renders a RadioGroup', () => {
    const { container } = renderWithTheme(
      <RadioGroup>
        <Radio value="radio-test" />
      </RadioGroup>,
    );
    const radioGroup = container.firstChild;

    expect(radioGroup).toBeInTheDocument();
    expect(radioGroup).toMatchSnapshot();
  });
});
