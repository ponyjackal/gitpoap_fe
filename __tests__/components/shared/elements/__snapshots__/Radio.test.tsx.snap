// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders a RadioGroup 1`] = `
.c0 .mantine-Radio-radio {
  -webkit-transition: 150ms border-color ease;
  transition: 150ms border-color ease;
  background-color: #1E1F2E;
  border: 0.0625rem solid #292B3D;
}

.c0 .mantine-Radio-radio::before {
  -webkit-transition: 150ms background-color ease;
  transition: 150ms background-color ease;
  background-color: #307AE8;
}

.c0 .mantine-Radio-radio:hover:not(.mantine-Radio-radioDisabled) {
  cursor: pointer;
}

.c0 .mantine-Radio-label {
  -webkit-transition: 150ms color ease;
  transition: 150ms color ease;
  font-family: PT Mono;
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
}

.c0 .mantine-Radio-label:hover:not(.mantine-Radio-labelDisabled) {
  color: #5596F6;
  cursor: pointer;
}

.c0 .mantine-Radio-label:hover:not(.mantine-Radio-labelDisabled) .mantine-Radio-radio {
  border-color: #5596F6;
}

.c0 .mantine-Radio-label:active:not(.mantine-Radio-labelDisabled) {
  color: #1761CF;
}

.c0 .mantine-Radio-label:active:not(.mantine-Radio-labelDisabled) .mantine-Radio-radio {
  border-color: #1761CF;
}

.c0 .mantine-Radio-label:active:not(.mantine-Radio-labelDisabled) .mantine-Radio-radio::before {
  background-color: #1761CF;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 20px;
  padding-top: 10px;
}

.emotion-4>* {
  box-sizing: border-box;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-tap-highlight-color: transparent;
}

.emotion-7 {
  position: relative;
}

.emotion-9 {
  -webkit-tap-highlight-color: transparent;
  background-color: #fff;
  border: 1px solid #ced4da;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  width: 20px;
  height: 20px;
  border-radius: 20px;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  transition-property: background-color,border-color;
  transition-timing-function: ease;
  transition-duration: 100ms;
  cursor: default;
}

.emotion-9:focus {
  outline-offset: 2px;
  outline: 2px solid #339af0;
}

.emotion-9:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9:checked {
  background: #228be6;
  border-color: #228be6;
}

.emotion-9:checked+.__mantine-ref-icon {
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-9:disabled {
  border-color: #ced4da;
  background-color: #f1f3f5;
}

.emotion-9:disabled+.__mantine-ref-icon {
  color: #ced4da;
}

.emotion-11 {
  color: #fff;
  opacity: 0;
  -webkit-transform: scale(0.75) translateY(2px);
  -moz-transform: scale(0.75) translateY(2px);
  -ms-transform: scale(0.75) translateY(2px);
  transform: scale(0.75) translateY(2px);
  -webkit-transition: opacity 100ms ease;
  transition: opacity 100ms ease;
  pointer-events: none;
  width: 8px;
  height: 8px;
  position: absolute;
  top: calc(50% - 4px);
  left: calc(50% - 4px);
}

<div
  class="emotion-0 emotion-1 c0 emotion-2"
>
  <div
    class="emotion-3 emotion-4"
    role="radiogroup"
  >
    <div
      class="emotion-5 emotion-6"
    >
      <div
        class="emotion-7 emotion-8"
      >
        <input
          class="emotion-9 emotion-10"
          id="mantine-r0"
          type="radio"
          value="radio-test"
        />
        <svg
          aria-hidden="true"
          class="__mantine-ref-icon emotion-11 emotion-12"
          fill="none"
          viewBox="0 0 5 5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 2.5a2.5 2.5 0 115 0 2.5 2.5 0 01-5 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
